epoch_multinode_test.go

```diff
@@ -259,6 +302,12 @@ func (c *testComm) ListNodes() []NodeID {
 }
 
 func (c *testComm) SendMessage(msg *Message, destination NodeID) {
+	c.lock.RLock()
```

Comment:
I assume the lock is needed for the message filter. 

Is it possible to just lock around it? and not use a defer.

We have a channel send below the function and I generally don't like sending into channels while holding a lock because the thread that reads from the channel might need the lock to progress. Not speaking about this current code but in general, no reason to hold an RLock for long if there is no need for it.